### PR TITLE
feat(TextArea): New TextArea Component

### PR DESCRIPTION
**Pull request by fysheets**
New bootstrap form-control TextArea component - leverages Paragon asInput

Note* based on: https://github.com/edx/paragon/tree/master/src/InputText

Ticket: [EDUCATOR-1824](https://openedx.atlassian.net/browse/EDUCATOR-1824)

### CODE REVIEW

**Comment by fysheets** [comment]
FYI @edx/educator-dahlia @dsjen @arizzitano - based on the ticket title uncertainty, I can rename this to InputArea if we want.  Also this autmatically sets width to 100% with the bootstrap styling (same as InputText) so for reference if we want the width to be less, we should use bootstrap's grid/col styles. Also this means that resizing will be specifically for height, not height & width unless we overwrite the styles explicitly. See bootstrap example here, https://getbootstrap.com/docs/4.0/components/forms/#form-controls

File: src/TextArea/index.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      onChange={props.onChange}
+      onBlur={props.onBlur}
+      aria-invalid={!props.isValid}
+      disabled={props.disabled}
+      required={props.required}
+    />
+  );
+}
+
+Text.propTypes = inputProps;
```
Comment (jaebradley) [inline src/TextArea/index.jsx]: Were you thinking about adding `propTypes` for props here that aren't covered by `inputProps` (for example, `isValid`)?

I think you could do something like

```javascript
Text.propTypes = Object.assign({}, inputProps, {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  placeholder: PropTypes.string,
  describedBy: PropTypes.string,
  isValid: PropTypes.bool,
  ...
});
```
but there could be a better way

File: src/TextArea/index.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import asInput, { inputProps } from '../asInput';
+
+function Text(props) {
```
Comment (jaebradley) [inline src/TextArea/index.jsx]: If you want to go down the syntactic sugar rabbit hole, you could write this as 

```javascript
const Text = props => (<textarea  ...props />)
```

and save a couple lines - obviously not a big deal either way

File: src/TextArea/TextArea.stories.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+       mollit anim id est laborum."
+    />
+  ))
+  .add('validation', () => (
+    <TextArea
+      name="username"
+      label="Username"
+      description="The unique name that identifies you throughout the site."
+      validator={(value) => {
+        let feedback = { isValid: true };
```
Comment (jaebradley) [inline src/TextArea/TextArea.stories.jsx]: Not a big deal but why not
```javascript
if (value.length < 3) {
  return { }
}

return { }
```
I get the single exit point argument but it does seem a lot simpler than dealing with an (IMHO) unnecessary variable assignment.

**Comment by jaebradley** [comment]
@fysheets don't forget to use [the Angular commit convention](https://gist.github.com/stephenparish/9941e89d80e2bc58a153#format-of-the-commit-message) particularly for the __merged commit message__ as the [`semantic-release` package uses the commit message format to inform package version upgrades](https://github.com/semantic-release/semantic-release#default-commit-message-format).

File: src/TextArea/index.jsx (hunk 1 above)
Comment (fysheets) [reply src/TextArea/index.jsx]: I kind of have the same answer for all of your feedback haha -- I used the InputText component as the base for the creation of this component, so the syntax choices essentially match up to that. If we think these changes should be made I'm happy to make them, but if not I'm also happy to have the two Components share similarities for uniformity sake.

**Comment by fysheets** [comment]
@jaebradley thanks for the links! I was going to find you prior to any merging to figure out if we had documentation of like a step by step guide of "the proper way to commit/merge/push in paragon" or anything.  I'll likely still snag you to ensure process is done correctly.

**Pull request merged**